src/zenml/client.py

```diff
@@ -2420,24 +2394,20 @@ def deploy_stack_component(
                 component_type == StackComponentType.EXPERIMENT_TRACKER
                 and flavor == "mlflow"
             ):
-                mlflow_bucket = outputs.get("mlflow-bucket")
-                if mlflow_bucket:
+                if mlflow_bucket := outputs.get("mlflow-bucket"):
                     logger.info(
                         "The bucket used for MLflow is: %s "
                         "You can use this bucket as an artifact store to "
                         "avoid having to create a new one.",
                         mlflow_bucket,
                     )
 
-            # if the cloud is k3d, then check the container registry
-            # outputs. If they are set, then create one.
-            if cloud == "k3d":
-                container_registry_outputs = {
+            if container_registry_outputs := {
                     k: v
                     for k, v in outputs.items()
                     if k.startswith("container_registry")
-                }
-                if container_registry_outputs:
+                }:
+                if cloud == "k3d":
```

Comment:
This is the only change I don't agree with in this PR, `if cloud == "k3d":` should stay the first check since it's much cheaper than looping over the container registry outputs

Reply:
issues updated with the next commit as you mentioned